src/lib/loaders/EngineLoader.ts

```diff
@@ -0,0 +1,14 @@
+import {EnginePlugin} from '@salesforce/code-analyzer-engine-api';
+import {SampleEnginePlugin} from '../../tmp/SampleEnginePlugin';
+
+export interface EngineLoader {
```

Comment:
Just a stylistic thing... since I've never heard of a Loader Pattern. This essentially is the Provider or Factory pattern and this loaders look like a Provider or Factory to me. (Note the only difference between a Provider and a Factory is that a Provider has a get method that takes no arguments and a Factory has a create method which usually takes arguments).

So consider either:

    export interface EnginePluginsProvider {
        get(): EnginePlugin[]
    }

or

    export interface EnginePluginsFactory {
        create(): EnginePlugin[]
    }